bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/RollershutterValue.java

```diff
@@ -85,43 +110,65 @@ public Command parseCommand(Command command) throws IllegalArgumentException {
                 return UpDownType.UP;
             } else if (updatedValue.equals(downString)) {
                 return UpDownType.DOWN;
-            } else if (updatedValue.equals(stopString)) {
+            } else if (updatedValue.equals(stopCommandString)) {
                 return StopMoveType.STOP;
             }
         }
         throw new IllegalStateException("Cannot call parseCommand() with " + command.toString());
     }
 
+    @Override
+    public Command parseCommand(Command command) throws IllegalArgumentException {
+        return parseType(command, upCommandString, downCommandString);
+    }
+
+    @Override
+    public Command parseMessage(Command command) throws IllegalArgumentException {
+        command = parseType(command, upStateString, downStateString);
+        if (inverted && command instanceof PercentType percentType) {
+            return new PercentType(100 - percentType.intValue());
+        }
+        return command;
+    }
+
     @Override
     public String getMQTTpublishValue(Command command, @Nullable String pattern) {
-        final String upString = this.upString;
-        final String downString = this.downString;
-        final String stopString = this.stopString;
+        return getMQTTpublishValue(command, transformExtentsToString);
+    }
+
+    public String getMQTTpublishValue(Command command, boolean transformExtentsToString) {
+        final String upCommandString = this.upCommandString;
+        final String downCommandString = this.downCommandString;
+        final String stopCommandString = this.stopCommandString;
         if (command == UpDownType.UP) {
-            if (upString != null) {
-                return upString;
+            if (upCommandString != null) {
+                return upCommandString;
             } else {
-                return ((UpDownType) command).name();
+                return (inverted ? "100" : "0");
```

Comment:
Could you define these "100" and "0" as constants with clear names to make it easier to understand that Up without inversion is zero.